paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc

```diff
@@ -25,28 +25,26 @@ namespace operators {
 using Tensor = framework::Tensor;
 using framework::DataLayout;
 
-template <typename T>
-class ConvTransposeMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+template <typename T, typename K, typename T_out>
+class ConvTransposeMKLDNNHandlerT
+    : public platform::MKLDNNHandlerT<T, mkldnn::deconvolution_forward> {
  public:
-  void Compute(const paddle::framework::ExecutionContext& ctx) const override {
-    PADDLE_ENFORCE_EQ(platform::is_cpu_place(ctx.GetPlace()), true,
-                      paddle::platform::errors::PreconditionNotMet(
-                          "Operator DNNL ConvTranspose must use CPUPlace"));
+  ConvTransposeMKLDNNHandlerT(const paddle::framework::ExecutionContext& ctx,
+                              const platform::MKLDNNDeviceContext& dev_ctx,
+                              const mkldnn::engine mkldnn_engine,
+                              platform::Place cpu_place, const Tensor* input,
+                              const Tensor* filter, const Tensor* bias,
+                              Tensor* output, const std::string& unique_name)
+      : platform::MKLDNNHandlerT<T, mkldnn::deconvolution_forward>(
+            dev_ctx, mkldnn_engine, cpu_place,
+            platform::CreateKey(dev_ctx, framework::vectorize(input->dims()),
+                                unique_name)) {
     const bool is_test = ctx.Attr<bool>("is_test");
```

Comment:
In general I like very much this PR . The only thing that is missing is that inside a ConvTransposeMKLDNNHandler you should call isCached() method not to create MD again. Please look to other ops implemented with MKLDNNHandlerT like pool, softmax etc.